src/_gettsim/transfers/rente.py

```diff
@@ -68,40 +164,121 @@ def ges_rente_vor_grundr_m(
     return out
 
 
-@policy_info(end_date="2020-12-31")
-def ges_rente_m(ges_rente_vor_grundr_m: float) -> float:
-    return ges_rente_vor_grundr_m
+@policy_info(end_date="2022-12-31")
+def bruttorente_mit_einkommensanrechnung_m(  # noqa: PLR0913
+    bruttorente_ohne_einkommensanrechnung_m: float,
+    bruttorente_nach_teilweiser_einkommensanrechnung_m: float,
+    bruttorente_nach_voller_einkommensanrechnung_m: float,
+    bruttolohn_m: float,
+    höchster_bruttolohn_letzte_15_jahre_m: float,
+    ges_rente_params: dict,
+) -> float:
+    """Pension benefit deduction due to earnings test for early retirement.
 
+    Deductions are made if earnings are above an earnings limit. If sum of earnings and
+    pension is larger than the highest income in the last 15 years, the pension is fully
+    deducted (Hinzuverdienstdeckel).
 
-@policy_info(
-    start_date="2021-01-01",
-    name_in_dag="ges_rente_m",
-    params_key_for_rounding="ges_rente",
-)
-def ges_rente_m_nach_grundr(
-    ges_rente_vor_grundr_m: float,
-    grundr_zuschlag_m: float,
-    rentner: bool,
+    Parameters
+    ----------
+    bruttorente_ohne_einkommensanrechnung_m
+        See :func:`bruttorente_ohne_einkommensanrechnung_m`.
+    bruttorente_nach_teilweiser_einkommensanrechnung_m
+        See :func:`bruttorente_nach_teilweiser_einkommensanrechnung_m`.
+    bruttorente_nach_voller_einkommensanrechnung_m
+        See :func:`bruttorente_nach_voller_einkommensanrechnung_m`.
+    bruttolohn_m
+        See basic input variable :ref:`bruttolohn_m <bruttolohn_m>`.
+    höchster_bruttolohn_letzte_15_jahre_m
+        See :func:`höchster_bruttolohn_letzte_15_jahre_m`.
+    ges_rente_params
+        See params documentation :ref:`ges_rente_params <ges_rente_params>`.
+
+    Returns
+    -------
+
+    """
+    if bruttolohn_m <= ges_rente_params["hinzuverdienstgrenze"] / 12:
+        out = bruttorente_ohne_einkommensanrechnung_m
+    # Teilrente unter Hinzuverdienstdeckel; 40% deduction rate
+    elif (
+        bruttolohn_m + bruttorente_ohne_einkommensanrechnung_m
+        <= höchster_bruttolohn_letzte_15_jahre_m
+    ):
+        out = bruttorente_nach_teilweiser_einkommensanrechnung_m
```

Comment:
I think the calculations are simple enough to include them here, more time-consuming to look all functions up than helpful to think at different levels. But I do not have a terribly strong opinion.